docs/tutorials/gallery_custom_components.py

```diff
@@ -0,0 +1,415 @@
+)
+
+# %%
+# And that's it. From here on you can use your custom components exactly like the
+# builtin ones. We are keeping it short here, but have a look at the
+# [Python API tutorial](../../generated/tutorials/gallery_python_api.md) for more
+# details.
+
+_ = await chat.prepare()
+
+# %%
+
+answer = await chat.answer("What is Ragna?")
+
+# %%
+
+print(answer)
+
+# %%
+for idx, source in enumerate(answer.sources, 1):
+    print(f"{idx}. {source.content}")
+
+# %%
+# ### REST API
+#
+# To be able to use custom components in Ragnas REST API or web UI, two things need to
+# happen:
+#
+# 1. We need move the code into a Python module or package, e.g. `tutorial.py` that is
+#    on the
+#    [`PYTHONPATH`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH),
+#    such that the classes can be imported.
+# 2. We need to include the components in the corresponding arrays in the configuration
+#    file. If you don't have a `ragna.toml` configuration file, see the
+#    [config reference](../../references/config.md) on how to create one.
+#
+#    For example, if we put the `TutorialSourceStorage` and `TutorialAssistant`
+#    classes in a `tutorial.py` module, we can expand the `source_storages` and
+#    `assistants` arrays like this
+#
+#    ```toml
+#    source_storages = [
+#        ...
+#        "tutorial.TutorialSourceStorage"
+#    ]
+#    assistants = [
+#        ...
+#        "tutorial.TutorialAssistant"
+#    ]
+#    ```
+#
+#    See the [config reference](../../references/config.md#referencing-python-objects)
+#    for details.
+#
+# And that's it again. We are keeping it short here, but have a look at the
+# [REST API tutorial](../../generated/tutorials/gallery_rest_api.md) for more
+# details.
+
+from ragna.deploy import Config
+
+config = Config(
+    source_storages=[TutorialSourceStorage],
+    assistants=[TutorialAssistant],
+)
+
+rest_api = ragna_docs.RestApi()
+
+client, document = rest_api.start(config, authenticate=True, upload_document=True)
+
+# %%
+# To select our custom components, we pass their display names to the chat creation.
+#
+# !!! tip
+#
+#     By default [ragna.core.Component.display_name][] returns the name of the class,
+#     but can be overridden, e.g. to format the name better.
+
+import json
+
+response = client.post(
+    "/chats",
+    json={
+        "name": "Tutorial REST API",
+        "documents": [document],
+        "source_storage": TutorialSourceStorage.display_name(),
+        "assistant": TutorialAssistant.display_name(),
+        "params": {},
+    },
+).raise_for_status()
+chat = response.json()
+
+client.post(f"/chats/{chat['id']}/prepare").raise_for_status()
+
+response = client.post(
+    f"/chats/{chat['id']}/answer",
+    json={"prompt": "What is Ragna?"},
+).raise_for_status()
+answer = response.json()
+print(json.dumps(answer, indent=2))
+
+# %%
+# Let's stop the REST API and have a look at what would have printed in the terminal if
+# we had started it the regular way.
```

Comment:
I think we should clearly state what the _regular way_ is (i.e. say that instead of regular), since it's unclear to a reader here (e.g. me!).

Do you mean through the CLI's `api` command here? If so, let's state that here.